app/jobs/data_warehouse/daily_sensitive_column_job.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module DataWarehouse
+  class DailySensitiveColumnJob < BaseJob
+    REPORT_NAME = 'daily-sensitive-column-job'
+
+    def perform(timestamp)
+      data = fetch_columns
+      upload_to_s3(data, timestamp)
+    end
+
+    def fetch_columns
+      tables = ActiveRecord::Base.connection.tables - %w[schema_migrations ar_internal_metadata]
+
+      sensitive_hash = []
+      insensitive_hash = []
+
+      tables.each do |table|
+        true_sensitives, false_sensitives = ActiveRecord::Base.connection.columns(table).
+          reject { |col| col.name == 'id' }.partition do |column|
+          column.comment&.match?(/sensitive=true/i)
+        end
+        insensitive_hash.concat(generate_column_data(false_sensitives, table))
+        sensitive_hash.concat(generate_column_data(true_sensitives, table))
+      end
+      {
+        sensitive: dasherize_keys(sensitive_hash),
+        insensitive: dasherize_keys(insensitive_hash),
+      }.to_json
+    end
+
+    def generate_column_data(columns, table)
+      columns.map do |column|
+        {
+          object_locator: {
+            column_name: column.name,
+            table_name: table,
+          },
+        }
+      end
+    end
+
+    def bucket_name
+      IdentityConfig.store.s3_idp_internal_dw_tasks
+    end
+
+    def dasherize_keys(columns)
+      columns.each do |column|
+        column.deep_transform_keys! { |key| key.to_s.dasherize }
+      end
+    end
```

Comment:
1. is there a specific convention we're matching with dasherized keys
2. since the keys are hardcoded a few lines above, what about just harcoding them as dasherized?

Reply:
Yep... we'll fix this.